src/main/java/uk/gov/companieshouse/accountsDates/AccountsDates.java

```diff
@@ -0,0 +1,249 @@
+     * 
+     */
+    public String generateBalanceSheetHeading(String periodStartString, String periodEndString, boolean isSameYear) {
+
+        Instant instantPeriodStart = Instant.parse(periodStartString);
+        Instant instantPeriodEnd = Instant.parse(periodEndString);
+        
+        LocalDate localDatePeriodStart = LocalDateTime.ofInstant(instantPeriodStart, ZoneId.of(ZoneOffset.UTC.getId())).toLocalDate();
+        LocalDate localDatePeriodEnd = LocalDateTime.ofInstant(instantPeriodEnd, ZoneId.of(ZoneOffset.UTC.getId())).toLocalDate();
+
+        Map<String, String> resultDates = calculatePeriodRange(localDatePeriodStart, localDatePeriodEnd, isSameYear);
+
+        if (!resultDates.containsKey(PERIOD_START)) {
+            return resultDates.get(PERIOD_END);
+        } else {
+            return resultDates.get(PERIOD_START) + " to " + resultDates.get(PERIOD_END);
+        }
+    }
+
+    /**
+     *Calculate balance sheet dates display format depending on range between period start and end dates
+     * 
+     * 
+     * @param periodStart accounting period start date
+     * @param periodEnd accounting period end date 
+     * @param isSameYear
+     * @return
+     * 
+     * 
+     */
+    public Map<String, String> calculatePeriodRange(LocalDate periodStart, LocalDate periodEnd, boolean isSameYear) {
+
+        int allowance = 15; // We allow year +/-15 days difference between dates to treat them as full year
+                            // difference
+        int yearRangeMax = 365 + allowance;
+        int yearRangeMin = 365 - allowance;
+
+        Map<String, String> periodObject = new HashMap<>();
+
+        // period returns years, hours and days between dates eg 1 year, 3 months and 4
+        // days
+        Period accountsPeriod = Period.between(periodStart, periodEnd);
+
+        long totalDaysDiff = ChronoUnit.DAYS.between(periodStart, periodEnd) + 1; // + 1 as they have time till the end
+                                                                                  // of the day
+        long totalMonthsDiff = ChronoUnit.MONTHS.between(periodStart, periodEnd);
+
+        // If the days over an exact month are 1 to 15, then round down (unless
+        // the period is less than 1 month, in which case round up).
+        // If the days over are 16 to 31, then round up. e.g. 15 months and 4 days is
+        // shown as 15 months; 15 months and 21 days is shown as 16 months.
+        totalMonthsDiff = (accountsPeriod.getDays() >= 15) ? totalMonthsDiff + 1 : totalMonthsDiff; // + 1 to days to
+                                                                                                    // include end date
+
+        // If the previous and current periods both end in the same year, then the
+        // heading is output as a full date e.g. ‘5 January 2015’ ’31 December 2015’.
+        if (isSameYear) {
+            periodObject.put(PERIOD_END, convertStringToDisplayDate(convertDateToString(periodEnd)));
```

Comment:
Can you change the convertStringToDisplayDate() method to take a LocalDate as a parameter instead of two convert methods being called?